filament/src/materials/colorGrading/colorGrading.mat

```diff
@@ -65,66 +75,104 @@ fragment {
 
 #include "../../../../shaders/src/dithering.fs"
 #include "../../../../shaders/src/vignette.fs"
-#include "../../../../shaders/src/bloom.fs"
-
-    vec3 colorGrade(mediump sampler3D lut, const vec3 x) {
-        // Alexa LogC EI 1000
-        const float a = 5.555556;
-        const float b = 0.047996;
-        const float c = 0.244161 / log2(10.0);
-        const float d = 0.386036;
-        vec3 logc = c * log2(a * x + b) + d;
-        return textureLod(lut, logc, 0.0).rgb;
-    }
 
-    vec3 resolveFragment(const ivec2 uv) {
-        return texelFetch(materialParams_colorBuffer, uv, 0).rgb;
-    }
+void dummy(){ }
 
-    vec4 resolveAlphaFragment(const ivec2 uv) {
-        return texelFetch(materialParams_colorBuffer, uv, 0);
+float starburst(const vec2 uv) {
+    // get an offset that continuously moves with the camera
+    vec3 forward = getViewFromWorldMatrix()[2].xyz;
+    float offset = forward.x + forward.y + forward.z;
+
+    vec2 center = uv - vec2(0.5001);
+    float d = length(center);
+    float radial = acosFastPositive(center.x / d);
+    float mask =
+        textureLod(materialParams_starburstBuffer, fract(vec2(radial + offset * 1.0, 0.0)), 0.0).r *
+        textureLod(materialParams_starburstBuffer, fract(vec2(radial - offset * 0.5, 0.0)), 0.0).r ;
+
+    return saturate(mask + (1.0 - smoothstep(0.0, 0.3, d)));
+}
+
+vec3 bloom(const vec3 color) {
+    highp vec2 uv = variable_vertex.xy;
+
+    vec3 result = vec3(0.0);
+
+    vec3 bloom = textureLod(materialParams_bloomBuffer, uv, 0.0).rgb;
+    result += bloom * materialParams.bloom.x;
+
+    float starburstMask = starburst(uv);
```

Comment:
Wait the lens flare cannot be disabled?

Reply:
and the starburst too.

Reply:
The flare can be disabled, but not the starburst (when the flare is active).

Reply:
Oh I see what you mean. Yes, I can probably do better here. I was just relying on the texture being a 1x1. But we can save all that ALU.

Good point.

Reply:
Yeah and I would also be careful with the texture unit cache. Given that color grading samples the 3D LUT + the color buffer + the bloom buffer + maybe the dirt buffer I don't know how much will actually fit in the cache? Anyway it seems worth doing a couple of uniform checks to avoid the texture fetches anyway + all the ALUs indeed (just like we do with the dirt texture basically).